### DISCUSSION TITLE
CommandStream used too much space.

### DISCUSSION THREAD

**Discussion by HappySapeta** [google/filament#7968] [Q&A]
I've written a simple program to test my filament integration.
There are no errors or major warnings in my builds, however the console gets spammed by the following message everytime I run my program.

`CommandStream used too much space (will block): needed space 33554432 out of 33554416, totalUsed=67108880, current=48, queue size=860334 buffers`

I discovered this [thread](https://github.com/google/filament/discussions/5427) where it was suggested to increase the minimum command buffer size by setting a higher value for FILAMENT_MIN_COMMAND_BUFFERS_SIZE_IN_MB.
No matter what value I try (I've tried 2, 4, and 32), the warnings do not go away.

There is just one renderable in my scene, Suzanne, provided by Filament. (These warnings appear even with a one triangle in the scene.)
There is no custom code apart from a GLFW integration.

What can I try to alleviate this issue ?

**Comment by korfriend** [comment]
I am suffering from the same issue. Additionally, when there are multiple cameras (that are not used for rendering) besides the main camera in Vulkan backend, errors occur in the Vulkan thread during program termination.

**Reply by HappySapeta** [reply]
Did you manage to fix this ?

**Reply by korfriend** [reply]
workaround : add a sleep thread to avoid such command stream overflow issues.
in my case, 
```
if (renderer->beginFrame(sc)) {
 renderer->render(view);
 renderer->endFrame();
}
std::this_thread::sleep_for(std::chrono::milliseconds(1));
 ```

**Comment by BStringhamVRSK** [comment]
You can also solve this problem by adding a fence at the end of your frame. Then, wait for the fence to clear, which indicates that the command buffer has been processed for that frame, before submitting another frame.
We typically have 2 frames "in flight," so we alternate between a couple of fences to achieve this, which allows one frame to be built while another frame is being processed by the GPU.
The functions you want are these (plus possibly other related functions):
```
Engine::createFence()
Fence::wait()
```
